notebooks/generative/mnist_autoencoder_using_kino.livemd

```diff
@@ -0,0 +1,262 @@
+# MNIST Autoencoder using Kino for visualization
+
+```elixir
+Mix.install([
+  {:exla, "~> 0.3.0"},
+  {:nx, "~> 0.3.0"},
+  {:axon, "~> 0.2.0"},
+  {:req, "~> 0.3.1"},
+  {:kino, "~> 0.6.2"}
+])
+```
+
+## Intro
+
+The goal of this notebook is to build a Autoencoder from scratch using Livebook. This notebook is based on [Training an Autoencoder on Fashion MNIST](https://github.com/hanrelan/axon/blob/main/notebooks/generative/fashionmnist_autoencoder.livemd), but includes some tips on using Livebook to train the model and using [Kino](https://hexdocs.pm/kino/Kino.html) (Livebook's interactive widget library) to play with and visualize our results.
+
+## Data loading
+
+An autoencoder learns to recreate it's seen in its dataset. For this notebook, we're going to try something simple: generating images of digits using the MNIST digit recognition dataset.
+
+<!-- livebook:{"break_markdown":true} -->
+
+To load the MNIST dataset, I'll use the Axon MNIST tutorial [here](https://github.com/elixir-nx/axon/blob/main/notebooks/mnist.livemd). The steps below are copied over from that tutorial
+
+```elixir
+mnist_base_url = "https://storage.googleapis.com/cvdf-datasets/mnist/"
+%{body: train_images} = Req.get!(mnist_base_url <> "train-images-idx3-ubyte.gz")
+# We're not going to use the labels so we won't grab them
+# %{body: train_labels} = Req.get!(base_url <> "train-labels-idx1-ubyte.gz")
+
+<<_::32, n_images::32, n_rows::32, n_cols::32, images::binary>> = train_images
+# Ignore the labels
+# <<_::32, n_labels::32, labels::binary>> = train_labels
+```
```

Comment:
Both this example and the one linked should probably use `SciData` since that has all of this logic built in :)